src/http.c

```diff
@@ -77,29 +101,42 @@ http_response_set_header(struct http_response *r, const char *k, const char *v)
 		if(strncmp(r->headers[i].key, k, key_sz) == 0) {
 			pos = i;
 			/* free old value before replacing it. */
-			free(r->headers[i].key);
-			free(r->headers[i].val);
+			if(r->headers[i].copy == HEADER_COPY_KEY) free(r->headers[i].key);
+			if(r->headers[i].copy == HEADER_COPY_VALUE) free(r->headers[i].val);
 			break;
 		}
 	}
```

Comment:
Looking at the callers of `http_response_set_header` it feels like all these calls to `strncmp` are probably useless for almost all responses, if not literally all. It could make sense to have a way to skip it if we know the header been added has never been set.